src/utils/actions.js

```diff
@@ -166,58 +165,64 @@ const baseIgActions = {
+
+          if (dateFilter) {
+            // assume two filters don't overlap
+            const path = dateFilter.path;
+            const searchParam = dateFilter.searchParam;
+            const filterPeriod = dateFilter.valuePeriod; // assume its a valuePeriod
+            if (path) {
+              filteredResources = filteredResources.filter(resource => {
+                // get value by path
+                const date = getByPath(resource, path); // assume its a datetime?
+                return compareDates(date, filterPeriod);
+              });
+            } else {
+              // path or searchParam are required
+              filteredResources = filteredResources.filter(resource => {
+                // get value by param
+                const date = getBySearchParam(resource, searchParam); // assume its a datetime?
+                return compareDates(date, filterPeriod);
+              });
+            }
           }
-        }
 
-        const codeFilter = input.codeFilter;
-        if (codeFilter) {
-          // code filter isn't under the same constraint as datefilter
-          // we can assume that it's just a path
-          const path = codeFilter.path;
-          const searchParam = codeFilter.searchParam;
-          if (path) {
-            // no need to refilter resources
-            filteredResources = filteredResources.filter(resource => {
-              const code = getByPath(resource, path);
-              // codeFilter.code is of type Coding, could also be a valueSet?
-              return compareCodes(code, codeFilter.code[0]);
-            });
-          } else if (searchParam) {
-            filteredResources = filteredResources.filter(resource => {
-              const code = getBySearchParam(resource, searchParam);
-              return compareCodes(code, codeFilter.code);
-            });
+          const codeFilter = input.codeFilter;
+          if (codeFilter) {
+            // code filter isn't under the same constraint as datefilter
+            // we can assume that it's just a path
+            const path = codeFilter.path;
+            const searchParam = codeFilter.searchParam;
+            if (path) {
+              // no need to refilter resources
+              filteredResources = filteredResources.filter(resource => {
+                const code = getByPath(resource, path);
+                // codeFilter.code is of type Coding, could also be a valueSet?
+                return compareCodes(code, codeFilter.code[0]);
+              });
+            } else if (searchParam) {
+              filteredResources = filteredResources.filter(resource => {
+                const code = getBySearchParam(resource, searchParam);
+                return compareCodes(code, codeFilter.code);
+              });
+            }
           }
-        }
-        return context.records.push(...filteredResources);
+
+          // QUESTION - should context have a mutex?
```

Comment:
I don't think it's necessary at this point. We aren't expecting to have multiple processes using the same object. If there are going to be 2 things looking at the same context it will be via the DB so not the same actual object

Reply:
Good point. Removing this comment